streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -516,6 +548,74 @@ public void close() {
     }
 
     @Test
+    public void shouldFlushAllStoresEvenIfStoreThrowsExcepiton() throws IOException {
+        final ProcessorStateManager stateManager = new ProcessorStateManager(
+            taskId,
+            Collections.singleton(changelogTopicPartition),
+            false,
+            stateDirectory,
+            Collections.singletonMap(storeName, changelogTopic),
+            changelogReader,
+            false,
+            logContext);
+
+        final MockStateStoreSupplier.MockStateStore stateStore1 = new MockStateStoreSupplier.MockStateStore(storeName, true) {
+            @Override
+            public void flush() {
+                throw new RuntimeException("KABOOM!");
+            }
+        };
+        final MockStateStoreSupplier.MockStateStore stateStore2 = new MockStateStoreSupplier.MockStateStore(storeName + "2", true) {
+            @Override
+            public void flush() {
+                flushedStore = true;
```

Comment:
This is only used by this test, so i think it would be better to make a local var `final AtomicBoolean flushedStore` and call `flushedStore.set(true)` here.

Reply:
I don't like the `AtomicBoolean` workaround to make it local... But ok.

Reply:
i think it is far cleaner than having static fields that are used only for the purpose of this test